crates/fuel-streams-publisher/src/publisher.rs

```diff
@@ -189,6 +177,26 @@ impl Publisher {
         &self.streams
     }
 
+    fn set_panic_hook(&mut self) {
+        let nats_client = self.nats_client.clone();
+        let fuel_service = Arc::clone(&self.fuel_service);
+        std::panic::set_hook(Box::new(move |panic_info| {
+            let payload = panic_info
+                .payload()
+                .downcast_ref::<&str>()
+                .unwrap_or(&"Unknown panic");
+            tracing::error!("Publisher panicked with a message: {:?}", payload);
+            let handle = tokio::runtime::Handle::current();
+            let nats_client = nats_client.clone();
+            let fuel_service = Arc::clone(&fuel_service);
+            handle.spawn(async move {
+                Publisher::flush_await_all_streams(&nats_client).await;
```

Comment:
Its a usual practice to at least attempt some cleanup. Usually if we die, we might just end up in a really bad state. Also , note that both methods flush_await_all_streams and stop_fuel do not throw exceptions, i.e. they cannot fail whatsoever.

Reply:
@lostman that happens because the owner process, `main` dies, right?  At that point, we're leveraging the auto-destructor from Tokio -- so graceful shutdown of processes still happens and we simply need our operations to be recoverable from panics.